mavutil.py

```diff
@@ -92,11 +92,11 @@ def __str__(self):
 
 def add_message(messages, mtype, msg):
     '''add a msg to array of messages, taking account of instance messages'''
-    if msg._instance_field is None or getattr(msg, msg._instance_field, None) is None:
+    if getattr(msg, '_instance_field', None) is None:
         # simple case, no instance field
         messages[mtype] = msg
         return
-    instance_value = getattr(msg, msg._instance_field)
+    instance_value = getattr(msg, '_instance_field')
```

Comment:
this breaks instance fields. Say for example msg._instance_field is "FOO" we want msg.FOO not msg._instance_field